### ISSUE TITLE
NAT-PMP/PCP daemon

### ISSUE THREAD

**Issue by stokito** [miniupnp/miniupnp#545]
I working on OpenWRT based distribution for a really small routers. I want to add PCP support but the whole miniupnp is too big and can't fit into my image. Is it possible to create a separate package with only PCP daemon?
Does it makes sense to create such service under your umbrella or I should try to create a separate project?

**Comment by stokito** [comment]
On my Turris router (arm) binary takes about 140k. This is extremely small but just for a comparison uhttpd takes 46k.
Probably that's why miniupnd is not installed on tiny images for 4mb devices.
Anyway, thank you, I'll try to play a little bit and investigate it.

**Comment by miniupnp** [comment]
@stokito is that stripped and built with -Os ?
You can disable a few feature, like removing `#define ENABLE_EVENTS` and `"#define ENABLE_NATPMP` from the config.h file

**Comment by miniupnp** [comment]
also disable IGDV2

**Comment by stokito** [comment]
Just an update on the topic for anybody interested.
I checked on OpenWrt with WR1043N  (MIPS) and after installing the miniupnpd and it's dependency libcap-ng the disk size usage increased on 72Kb. That's because the disk is compressed with LZMA. I think this is negligible even for 4Mb routers.

Just for a curiosity I tried to compile it on Ubuntu with disabling those flags that you mentioned and this minimized binary on 8%.
Off course it still contains the UPnP support.

Also I checked few programs and looks like the UPnP is still used but the NAT-PMP/PCP is not yet so widely deployed.
That means that having the UPnP support still makes some sense.

Also I found two NAT-PMP only servers:
* http://savannah.nongnu.org/projects/natpmp/ very old and simple written by a student.
* https://github.com/tedjp/stallone also outdated but looks more advanced.

Personally I think that it's not necessary to have the PCP only build.
I going to send a letter to OpenWrt devel list and ask them to include the miniupnpd by default into all OpenWrt builds except of tiny.
